impeller/renderer/backend/vulkan/command_encoder_vk.cc

```diff
@@ -1,209 +0,0 @@
-    VALIDATION_LOG << "Failed to end command buffer: " << vk::to_string(status);
-    return false;
-  }
-  return true;
-}
-
-vk::CommandBuffer CommandEncoderVK::GetCommandBuffer() const {
-  if (tracked_objects_) {
-    return tracked_objects_->GetCommandBuffer();
-  }
-  return {};
-}
-
-void CommandEncoderVK::Reset() {
-  tracked_objects_.reset();
-
-  queue_ = nullptr;
-  is_valid_ = false;
-}
-
-bool CommandEncoderVK::Track(std::shared_ptr<SharedObjectVK> object) {
-  if (!IsValid()) {
-    return false;
-  }
-  tracked_objects_->Track(std::move(object));
-  return true;
-}
-
-bool CommandEncoderVK::Track(std::shared_ptr<const DeviceBuffer> buffer) {
-  if (!IsValid()) {
-    return false;
-  }
-  tracked_objects_->Track(std::move(buffer));
-  return true;
-}
-
-bool CommandEncoderVK::IsTracking(
-    const std::shared_ptr<const DeviceBuffer>& buffer) const {
-  if (!IsValid()) {
-    return false;
-  }
-  return tracked_objects_->IsTracking(buffer);
-}
-
-bool CommandEncoderVK::Track(std::shared_ptr<const TextureSourceVK> texture) {
-  if (!IsValid()) {
-    return false;
-  }
-  tracked_objects_->Track(std::move(texture));
-  return true;
-}
-
-bool CommandEncoderVK::Track(const std::shared_ptr<const Texture>& texture) {
-  if (!IsValid()) {
-    return false;
-  }
-  if (!texture) {
-    return true;
-  }
-  return Track(TextureVK::Cast(*texture).GetTextureSource());
-}
-
-bool CommandEncoderVK::IsTracking(
-    const std::shared_ptr<const Texture>& texture) const {
-  if (!IsValid()) {
-    return false;
-  }
-  std::shared_ptr<const TextureSourceVK> source =
-      TextureVK::Cast(*texture).GetTextureSource();
-  return tracked_objects_->IsTracking(source);
-}
-
-fml::StatusOr<vk::DescriptorSet> CommandEncoderVK::AllocateDescriptorSets(
-    const vk::DescriptorSetLayout& layout,
-    const ContextVK& context) {
-  if (!IsValid()) {
-    return fml::Status(fml::StatusCode::kUnknown, "command encoder invalid");
-  }
-
-  return tracked_objects_->GetDescriptorPool().AllocateDescriptorSets(layout,
-                                                                      context);
-}
-
-void CommandEncoderVK::PushDebugGroup(std::string_view label) const {
-  if (!HasValidationLayers()) {
-    return;
-  }
-  vk::DebugUtilsLabelEXT label_info;
-  label_info.pLabelName = label.data();
-  if (auto command_buffer = GetCommandBuffer()) {
-    command_buffer.beginDebugUtilsLabelEXT(label_info);
-  }
-}
-
-void CommandEncoderVK::PopDebugGroup() const {
-  if (!HasValidationLayers()) {
-    return;
-  }
-  if (auto command_buffer = GetCommandBuffer()) {
-    command_buffer.endDebugUtilsLabelEXT();
-  }
-}
-
-void CommandEncoderVK::InsertDebugMarker(std::string_view label) const {
-  if (!HasValidationLayers()) {
-    return;
-  }
-  vk::DebugUtilsLabelEXT label_info;
-  label_info.pLabelName = label.data();
-  if (auto command_buffer = GetCommandBuffer()) {
-    command_buffer.insertDebugUtilsLabelEXT(label_info);
-  }
-  if (queue_) {
```

Comment:
TBD: what was this for?

Reply:
Asking because if I remove this, the command buffer no longer needs a reference to the queue.

Reply:
I was inserting the same marker in both the queue for the buffer for ease of debugging. But AFAICT, having the marker (push-pop) in the buffer is sufficient. This complicates other things like reset. So let's get rid of it.